sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -2229,6 +2229,12 @@
         } else if (entry.getContent() == null) {
             logger.info("entry.getContent() is null. The entity may not exist. {}", entry);
             return new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), null);
+        } else if (entry.getContent().getQueueDescription() == null) {
```

Comment:
Is there a user story for this? I feel like it is a user error if we are expecting a Queue and it returns a Topic instead and I would expect an error.

Reply:
The issue [19512](https://github.com/Azure/azure-sdk-for-java/issues/19512) occurs when a queue exists but the name is used for a topic **and vice versa**.
Vice versa means that it occurs also when a topic exists but the name is used for a queue.

Therefore I made the same fix for the queue deserialization ([here](https://github.com/Azure/azure-sdk-for-java/pull/19513/files/457d911af6e128fb7d963151f99d2d748e79e167#diff-8b1de19ce3e7548d62bea61e56b8d6dd9e11eeb8951f3afc1b5b09b9cf798e5cR2232)) and topic deserialization ([here](https://github.com/Azure/azure-sdk-for-java/pull/19513/files/457d911af6e128fb7d963151f99d2d748e79e167#diff-8b1de19ce3e7548d62bea61e56b8d6dd9e11eeb8951f3afc1b5b09b9cf798e5cR2317)).